src/System.Runtime.Extensions/tests/System/Environment.SetEnvironmentVariable.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Xunit;
+
+public class SetEnvironmentVariable
```

Comment:
I had the same thought.  It looks like we have a test to verify that ProcessStartInfo gets the right environment variables, but we don't verify the environment of the child process itself.